lib/AST/Type.cpp

```diff
@@ -2484,9 +2484,18 @@ bool ReplaceOpaqueTypesWithUnderlyingTypes::shouldPerformSubstitution(
 
   // Allow replacement of opaque result types of inlineable function regardless
   // of resilience and in which context.
-  if (namingDecl->getAttrs().hasAttribute<InlinableAttr>()) {
-    return true;
+  if (auto *afd = dyn_cast<AbstractFunctionDecl>(namingDecl)) {
```

Comment:
Can you add some tests where either the property itself or just the getter is inlinable?

Reply:
There is one. It failed when I had not added the code for abstract storage decl.

https://github.com/apple/swift/blob/master/test/SILOptimizer/specialize_opaque_type_archetypes.swift#L335

https://github.com/apple/swift/blob/master/test/SILOptimizer/Inputs/specialize_opaque_type_archetypes_3.swift#L59

Reply:
Ah ok. A case you might want to test that the old logic definitely would have missed is an `@_alwaysEmitIntoClient` decl. These are the same as inlinable except they have different linkage -- but don't have an InlinableAttr.